deepspeed/runtime/utils.py

```diff
@@ -870,12 +880,15 @@ def get_global_norm_of_tensors(input_tensors, norm_type=2, mpu=None, use_graph=F
     assert all([torch.is_tensor(t) for t in input_tensors]), f'expected list of only tensors'
 
     norm_type = float(norm_type)
+    all_norms = []
     if norm_type == inf:
-        total_norm = max(t.data.abs().max() for t in input_tensors)
-        total_norm_cuda = get_accelerator().FloatTensor([float(total_norm)])
+        for t in input_tensors:
+            all_norms.append(t.data.abs().max().float())
+        total_norm_cuda = torch.stack(all_norms).max()
```

Comment:
Rename to make device-agnostic
```suggestion
        device_total_norm = torch.stack(all_norms).max()
```

Reply:
done